org.testfx.service.finder.NodeFinderException not found in testfx-core-4.0.11-alpha, but required from testfx-legacy-4.0.8-alpha

## Expected Behavior
 We have a testsuite, that uses testfx. It works fine with 
       **testfx-core-4.0.10-alpha** and **testfx-legacy-4.0.8-alpha**

but fails with 
       **testfx-core-4.0.11-alpha** and **testfx-legacy-4.0.8-alpha**

We guess, the reason is, that **org.testfx.service.finder.NodeFinderException** is present
in **testfx-core-4.0.10-alpha.jar** but not contained in  **testfx-core-4.0.11-alpha.jar** (maybe other classes are also missing, we did not check that). 

Please either update legacy accordingly or patch core. We used
    **compile "org.testfx:testfx-core:4.+"
    compile "org.testfx:testfx-legacy:4.+"**
as gradle dependencies, which is, how we ran into this problem.

## Actual Behavior
 a lot of exceptions like:

java.lang.NoClassDefFoundError: org/testfx/service/finder/NodeFinderException
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at org.junit.internal.builders.SuiteMethodBuilder.hasSuiteMethod(SuiteMethodBuilder.java:18)
	at org.junit.internal.builders.SuiteMethodBuilder.runnerForClass(SuiteMethodBuilder.java:10)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy1.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:108)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: org.testfx.service.finder.NodeFinderException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 40 more

## Specifications

  - Version:      testfx-core-4.0.10-alpha and testfx-legacy-4.0.8-alpha
  - Platform: Windows 10 et.all

--- a/README.md
+++ b/README.md
@@ -217,6 +217,14 @@ class ClickableButtonSpec extends ApplicationSpec {
 
 Head over to our [gitter chat](https://gitter.im/TestFX/TestFX) for discussion and questions.
 
+## TestFX Legacy: Deprecated
+
+The `testfx-legacy` subproject is deprecated and no longer supported. It is highly recommended
+that you switch from using `testfx-legacy`. If you want to continue using it you should cap
+the versions of `testfx-core` and `testfx-legacy` to `4.0.8-alpha`, which was the last released
+version of `testfx-legacy`. Using a newer version of `testfx-core` with an older version of
+`testfx-legacy` will very likely break (and does with `testfx-core` versions past `4.0.10-alpha`).
+
 ## Credits
 
 Thanks to all of the [contributors of TestFX](https://github.com/TestFX/TestFX/graphs/contributors)!

